### DISCUSSION TITLE
Deploy monorepo to multiple instances?

### DISCUSSION THREAD

**Discussion by eivindml** [vercel/vercel#3919] [General] [answered]
Hi 👋🏼,
I have a monorepo with a structure like this:

```
monorepo/
  website1/
  website2/
  website3/
  ...
```

Is there a way to orchestrate the deploy from root and/or using the Github integration the deploy each website to it's own instance with it's own domain? 

I guess I could add a `now.json` to each `website/` folder and have a script in the root dir that loops through each folder and deploys them. But would love a more 'native' solution and support for "push to deploy" through GitHub.

🥳

**Comment by coetry** [comment, marked as answer]
We're *absolutely* going to enable this from the dashboard soon and you'll be able to deploy multiple projects from a single repo and give each project its own settings (all configured from the Web UI) :rocket:

**Reply by antoniojps** [reply]
That's awesome! Is there any alternative right now? Any ETA?

**Reply by ibraelillo** [reply]
@coetry when? I'm struggling with my monorepo  :(

**Reply by antoniojps** [reply]
I ended up having to use other services and only use Vercel for the next.js project 
 Hopefully this will be available soon.
😞

**Reply by reflash** [reply]
@coetry will it be possible to do the same via configuration files?

**Comment by dohomi** [comment]
I currently did this now version < 17 with the `now --name project1` `now --name project2` but after now v17 the name is deprecated and I am looking for a solution as well. I am running a NextJS project and Storybook in the same repo and would like to deploy the different projects from the same folder. What is currently the best solution? I am still using now v16 due to this reasons

**Reply by ryardley** [reply]
Would love to see a recipe or blog post outlining how to deploy Next.js + Storybook. Next.js -> `mysite.com` + Storybook -> `styleguide.mysite.com` with `now`

**Reply by antoniojps** [reply]
@ryardley I need that so much! Trying to find a solution.

**Reply by murillo94** [reply]
me too :)

**Comment by eivindml** [comment]
Which alternatives to achieve this do we have right now?

**Comment by reflash** [comment]
I have a similar case, one thing different is that if I try setting now.json in the nested folder and try to deploy it, I get an error. it's not able to import a workspace dependency.
it works if now.json is at the root, but means I cannot deploy different websites independently

**Comment by eivindml** [comment]
This is how I'm currently doing it, but would love to hear how this can be improved: https://github.com/vercel/vercel/discussions/4582
